hummingbot/connector/exchange/okx/okx_api_order_book_data_source.py

```diff
@@ -73,6 +73,25 @@ async def _request_order_book_snapshot(self, trading_pair: str) -> Dict[str, Any
 
         return data
 
+    async def _parse_order_book_snapshot_message(self, raw_message: Dict[str, Any], message_queue: asyncio.Queue):
```

Comment:
the idea is good, but in the implementation, you are just selecting the first one on the list. You should loop over the messages that you receive instead of indexing that by 0. Check how the diffs message is implemented

Reply:
Hi @cardosofede, this is the example response from the [reference](https://www.okx.com/docs-v5/en/#websocket-api-public-channel-order-book-channel), it can be expected that there is only 1 item in the data list for both diffs and snapshots. 
```
{
  "arg": {
    "channel": "books",
    "instId": "BTC-USDT"
  },
  "action": "snapshot",
  "data": [
    {
      "asks": [
        ["8476.98", "415", "0", "13"],
        ["8477", "7", "0", "2"],
        ["8477.34", "85", "0", "1"],
        ["8477.56", "1", "0", "1"],
        ["8505.84", "8", "0", "1"],
        ["8506.37", "85", "0", "1"],
        ["8506.49", "2", "0", "1"],
        ["8506.96", "100", "0", "2"]
      ],
      "bids": [
        ["8476.97", "256", "0", "12"],
        ["8475.55", "101", "0", "1"],
        ["8475.54", "100", "0", "1"],
        ["8475.3", "1", "0", "1"],
        ["8447.32", "6", "0", "1"],
        ["8447.02", "246", "0", "1"],
        ["8446.83", "24", "0", "1"],
        ["8446", "95", "0", "3"]
      ],
      "ts": "1597026383085",
      "checksum": -855196043
    }
  ]
}
```

Reply:
@cardosofede please also assign https://github.com/hummingbot/hummingbot/issues/5813 to me thanks

Reply:
Is the same issue that you are trying to address.
Yes I know that normally the response only includes one dictionary, but is better to loop to prevent empty messages and if there is more than one dictionary inside you will be covered too.

Reply:
btw, is just a recommendation, if it's tested and you never found the other behavior, it's ready to go! @JeremyKono

Reply:
Yes I have tested it live and added unit test. There was no bad behaviour for the expected response message.

Reply:
Thanks for the recommendation @cardosofede , I will keep it in mind.